graylog2-shared/src/main/java/org/graylog2/shared/buffers/JournallingMessageHandler.java

```diff
@@ -58,14 +62,30 @@ public void onEvent(RawMessageEvent event, long sequence, boolean endOfBatch) th
 
             final Converter converter = new Converter();
             // copy to avoid re-running this all the time
-            final List<Journal.Entry> entries = Lists.newArrayList(transform(batch, converter));
-            final long lastOffset = journal.write(entries);
-            log.debug("Processed batch, wrote {} bytes, last journal offset: {}, signalling reader.",
-                      converter.getBytesWritten(),
-                      lastOffset);
-            journalFilled.release();
+            // Remove all null values returned from the converter (might happen if the Converter throws an exception)
+            final List<Journal.Entry> entries = Lists.newArrayList(filter(transform(batch, converter), notNull()));
```

Comment:
Maybe using [`Collection#removeAll()`](https://docs.oracle.com/javase/7/docs/api/java/util/Collection.html#removeAll(java.util.Collection)) could make sense here instead of using `Collections2.filter()` to prevent additional copies.

```java
final List<Journal.Entry> entries = Lists.newArrayList(transform(batch, converter));
entries.removeAll(Collections.singleton(null));
```